src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingMethodSymbol.cs

```diff
@@ -86,6 +86,15 @@ public override MethodSymbol UnderlyingMethod
             }
         }
 
+        public override TypeSymbol ReceiverType
+        {
+            get
+            {
+                var receiver = _underlyingMethod.ReceiverType;
+                return this.RetargetingTranslator.Retarget(receiver, RetargetOptions.RetargetPrimitiveTypesByTypeCode);
```

Comment:
Is RetargetPrimitiveTypesByTypeCode the right option to use? Is this type going to be encoded in metadata by using type codes?